frontend/web/components/metadata/AddMetadataToEntity.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { FC, useEffect, useState } from 'react'
+import PanelSearch from 'components/PanelSearch'
+import Button from 'components/base/forms/Button'
+import Icon from 'components/Icon'
+import { useGetMetadataModelFieldListQuery } from 'common/services/useMetadataModelField'
+import { useGetMetadataFieldListQuery } from 'common/services/useMetadataField'
+import { useGetSegmentQuery } from 'common/services/useSegment'
+import { MetadataField, Metadata, ProjectFlag } from 'common/types/responses'
+import Input from 'components/base/forms/Input'
+import Utils from 'common/utils/utils'
+import { useGetProjectFlagQuery } from 'common/services/useProjectFlag'
+import Tooltip from 'components/Tooltip'
+import { sortBy } from 'lodash'
+
+type CustomMetadataField = MetadataField & {
+  metadataModelFieldId: number | string | null
+  isRequiredFor: boolean
+  model_field?: string | number
+}
+
+type CustomMetadata = (Metadata & CustomMetadataField) | null
+
+type AddMetadataToEntityType = {
+  organisationId: string
+  projectId: string | number
+  entityContentType: number
+  entityId: string
+  entity: number | string
+  createMetadataField: () => void
+  updateMetadata: () => void
```

Comment:
Are these types for updating / entering info to metadata? If so I think the types are incorrect?

Reply:
Removed